newswires/app/db/FingerpostWireEntry.scala

```diff
@@ -121,32 +67,36 @@ object FingerpostWireEntry
 
   def apply(
       fm: ResultName[FingerpostWireEntry]
-  )(rs: WrappedResultSet): FingerpostWireEntry = {
-    val fingerpostContent = Json.parse(rs.string(fm.content)).as[FingerpostWire]
-    val maybeCategoryCodes = rs.arrayOpt(fm.categoryCodes)
-    val categoryCodes = maybeCategoryCodes match {
-      case Some(array) =>
-        array.getArray
-          .asInstanceOf[Array[String]]
-          .toList
-      case None => Nil
+  )(rs: WrappedResultSet): Option[FingerpostWireEntry] = {
+    for {
+      fingerpostContent <- decode[FingerpostWire](
+        rs.string(fm.content)
+      ).toOption
```

Comment:
It's a bit awkward here to do error handling if there is an item in the database that isn't the correct json format. I know it should be difficult for entries in the database to not be of the correct format, but I also didn't want to make type assumptions in the code base. So I've changed the signature here to an option and any rogue dataitems would be quietly dropped in this case. But it means the option needs to be handled down stream which is a bit annoying.  

Could also do a try / catch and log out any bad data entries to preserve the old type signature? 

Open to suggestion